Mathlib/Data/Nat/Prime.lean

```diff
@@ -455,6 +455,21 @@ theorem exists_dvd_of_not_prime2 {n : ℕ} (n2 : 2 ≤ n) (np : ¬Prime n) :
     (not_prime_iff_minFac_lt n2).1 np⟩
 #align nat.exists_dvd_of_not_prime2 Nat.exists_dvd_of_not_prime2
 
+theorem not_prime_of_dvd (h : ∃ m, m ∣ n ∧ m ≠ 1 ∧ m ≠ n) : ¬Prime n := by
```

Comment:
Usually we would not package up all these hypotheses as an existential. It should just be `{m n} (h1 : m ∣ n) (h2 : ...) (h3 : ...)`.

Reply:
Done.